tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/graph/graph_component.ng.html

```diff
@@ -21,12 +21,121 @@
   </div>
 
   <div class="graph-structure-container">
-    <!-- TODO(cais): Add div for self op, input ops and consumer ops. -->
-    <!-- Make the following div "ng-template #noOpFocuse" of the else branch of an ngIf -->
+    <div *ngIf="opInfo !== undefined && opInfo !== null; else noOpFocused">
+      <div *ngIf="inputOps.length > 0; else noInputs" class="inputs-container">
+        <div>
+          <div
+            *ngFor="let inputOpInfo of inputOps; let slot = index"
+            class="input-op-section"
+          >
+            <div class="input-slot-header">
+              Input slot {{slot}}:
+            </div>
+            <div class="op-container input-op-container">
+              <div class="input-tensor-name">
+                <div
+                  class="op-name input-op-name"
+                  (click)="onGraphOpNavigate.emit({graph_id: graphId, op_name: inputOpInfo.op_name})"
+                >
+                  <span>{{inputOpInfo.op_name}}</span>
+                </div>
+                <div class="input-op-output-slot slot">
+                  Output slot: {{inputOpInfo.output_slot}}
+                </div>
+              </div>
+              <div
+                *ngIf="inputOpInfo.data !== undefined; else opInfoMissing"
+                class="op-type"
+              >
+                {{inputOpInfo.data.op_type}}
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
 
-    <div class="no-op-focused">
-      No graph op selected. Click a row in the Graph Executions table to view
-      the neighborhood of the underlying op in its graph.
+      <ng-template #noInputs>
+        <div class="inputs-container no-inputs-indicator">
+          (This op has no data input tensor.)
```

Comment:
`no data input tensor`, not sure if this is widely understandable. Just say `no input tensor` instead?